### DISCUSSION TITLE
gatsby-plugin-mdx v4 code in heading

### DISCUSSION THREAD

**Discussion by NickyMeuleman** [gatsbyjs/gatsby#36829] [Help] [answered]
I'm updating my gatsby theme to the latest gatsby-plugin-mdx and noticed a heading with inline code has a different entry in the `tableOfContents` compared to the older gatsby-plugin-mdx.

The heading is:
\## Shed your bikes, \`let` sleeping dogs lie

In the `tableOfContents` object you get from querying that mdx, the `title` is `" sleeping dogs lie"`.
(the older gatsby-plugin-mdx produced `"Shed your bikes, let sleeping dogs lie"`

Screenshot of rendered header with the correct CSS applied, and next to it a table of contents component that uses that wrong `title` field.
![chrome_iFTMl9JSJm](https://user-images.githubusercontent.com/30179461/196050934-1efcd3f2-8a4e-4692-8782-73c5e2338582.png)

The output from the `tableOfContents` object for that post:
```
{
  "items": [
    // ...
    {
      "url": "#useful-knowledge",
      "title": "Useful knowledge",
      "items": [
        {
          "url": "#shed-your-bikes-let-sleeping-dogs-lie",
          "title": " sleeping dogs lie"
        }
      ]
    },
    // ...
  ]
}
```

Why is this happening?
Can I code something to get complete text (that includes the text in the inline code)?

**Comment by LekoArts** [comment, marked as answer]
The ToC gets generated here: https://github.com/gatsbyjs/gatsby/blob/ab51cd446d7da41a6c1f5f9558bcf67d4e336a90/packages/gatsby-plugin-mdx/src/gatsby-node.ts#L224-L274

So it uses https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-mdx/src/remark-infer-toc-meta.ts and https://github.com/syntax-tree/mdast-util-toc under the hood. If you have time you could check out its code to see where it drops that information (if it's intentional or accidental)

My gut feeling would be that in e.g. https://github.com/gatsbyjs/gatsby/blob/ab51cd446d7da41a6c1f5f9558bcf67d4e336a90/packages/gatsby-plugin-mdx/src/remark-infer-toc-meta.ts#L48-L55 we don't account for `code`

**Reply by LekoArts** [reply]
@NickyMeuleman A solution should be here: https://github.com/gatsbyjs/gatsby/issues/37704
